### PR TITLE
exp init: add simple dvc exp init command

### PR DESCRIPTION
**Pull request by skshetry**
Simpler, initial implementation for `exp init`.

Closes https://github.com/iterative/dvc/issues/6441.
Part of https://github.com/iterative/dvc/issues/6446.

### CODE REVIEW

File: dvc/command/experiments.py [hunk 1]
```diff
@@ -789,6 +790,62 @@ def run(self):
...
+        name = self.args.name or self.DEFAULT_NAME
+        stage = self.repo.stage.add(
+            name=name,
+            cmd=cmd,
+            deps=[code, data],
+            outs=[models],
+            params=[{params_path: params}],
+            metrics_no_cache=[metrics],
+            plots_no_cache=[plots],
+            live=dvclive,
```
Comment (dberenbaum) [inline dvc/command/experiments.py:840]: Default stage can ignore `--dvclive` arg. We probably need conditional arg parsing based on the template, throwing an error if a provided argument is irrelevant for that template.

File: dvc/command/experiments.py [hunk 2]
```diff
@@ -1303,6 +1360,68 @@ def add_parser(subparsers, parent_parser):
     )
     experiments_remove_parser.set_defaults(func=CmdExperimentsRemove)
 
+    EXPERIMENTS_INIT_HELP = "Create experiments."
```
Comment (skshetry) [inline dvc/command/experiments.py:1363]:
```suggestion
    EXPERIMENTS_INIT_HELP = "Initialize experiments."
```

File: dvc/command/experiments.py (hunk 1 above)
Comment (skshetry) [reply dvc/command/experiments.py:840]: We don't have a default stage/template yet, so I left it with the default. I'll change this when we have templates.

File: dvc/command/experiments.py [hunk 3]
```diff
@@ -789,6 +790,62 @@ def run(self):
...
+        stage = self.repo.stage.add(
+            name=name,
+            cmd=cmd,
+            deps=[code, data],
+            outs=[models],
+            params=[{params_path: params}],
+            metrics_no_cache=[metrics],
+            plots_no_cache=[plots],
+            live=dvclive,
+            force=True,
```
[the remark this answers is not in the document]
Comment (skshetry) [reply dvc/command/experiments.py:841]: I'll add the `force` option internally on the next PR. :)

File: dvc/command/experiments.py [hunk 4]
```diff
@@ -1303,6 +1360,68 @@ def add_parser(subparsers, parent_parser):
...
+        help="Run the experiment after initializing it",
+    )
+    experiments_init_parser.add_argument(
+        "--interactive",
+        "-i",
+        action="store_true",
+        help="Prompt for values that are not provided",
+    )
+    experiments_init_parser.add_argument(
+        "--template", help="Stage template to use to fill with provided values"
```
[the remark this answers is not in the document]
Comment (skshetry) [reply dvc/command/experiments.py:1388]: Will add defaults in the upcoming PRs, as the place where I'll keep those defaults may constantly change. :)

**Comment by skshetry** [comment]
@dberenbaum, I am merging this and will follow up on your comments in the future PRs. The command is hidden meanwhile.

File: dvc/command/experiments.py [hunk 5]
```diff
@@ -1303,6 +1360,68 @@ def add_parser(subparsers, parent_parser):
...
+    )
+    experiments_init_parser.add_argument(
+        "--explicit", help="Only use the path values explicitly provided"
```
Comment (dberenbaum) [inline dvc/command/experiments.py:1391]: By the way, feel free to suggest a better name for this option. Maybe `--ignore-missing`?

**Pull request merged**